src/content/en/fundamentals/engage-and-retain/push-notifications/how-push-works.md

```diff
@@ -0,0 +1,160 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+
+{# wf_updated_on: 2017-03-03 #}
+{# wf_published_on: 2016-06-30 #}
+
+# How Push Works {: .page-title }
+
+{% include "web/_shared/contributors/mattgaunt.html" %}
+
+
+
+Before getting into the API, let's look at push from a highlevel, start to finish. Then as we
+step through individual topics or API's later on, you'll have an idea of how and why it's
+important.
+
+The three key steps when implementing push are:
+
+1. Adding the client side logic to subscribe a user to push (i.e. the JavaScript and UI in your
+web app that registers a user to push messages).
+1. The API call from your back-end / application that triggers a push message  to a users device.
+1. The service worker JavaScript file that will receive a "push event" when the push arrives on
+the device. It's in this JavaScript that you'll be able to show a notification.
+
+Let's look at what each of these steps entails in a little more detail.
+
+## Step 1: Client Side
+
+The first step is to "subscribe" a user to push messaging.
+
+Subscribing a user requires two things. First, getting **permission** from the user to send
+them push messages. Second, getting a `PushSubscription` from the browser.
+
+A `PushSubscription` contains all the information we need to send a push message to that user.
+You can "kind of" think of this as a ID for that user's device.
+
+This is all done in JavaScript with the [Push
+API](https://developer.mozilla.org/en-US/docs/Web/API/Push_API).
+
+Before subscribing a user you'll need to generate a set of
+"application server keys", which we'll cover later on.
+
+The application server keys, also known as VAPID keys, are unique to your server. They allow a
+push service to know which application server subscribed a user and ensure that it's the same
+server triggering the push messages to that user.
+
+Once you've subscribed the user and have a `PushSubscription`, you'll need to send the
+`PushSubscription` details to your backend / server.  On your server, you'll save this
+subscription to a database and use it to send a push message to that user.
+
+![Make sure you send the PushSubscription to your backend.](./images/svgs/browser-to-server.svg)
+
+## Step 2: Send a Push Message
+
+When you want to send a push message to your users you need to make an API call to a push
+service. This API call would include what data to send, who to send the message to and any
+criteria about how to send the message.
```

Comment:
Remember I (the reader) am new to push. Don't make me infer that you've changed contexts from client to server.